controlplane/kubeadm/internal/workload_cluster_etcd.go

```diff
@@ -130,6 +144,17 @@ func (w *Workload) EtcdIsHealthy(ctx context.Context) (HealthCheckResult, error)
 		return response, errors.Errorf("there are %d healthy etcd pods, but %d etcd members", expectedMembers, len(knownMemberIDSet))
 	}
 
+	// If the member list view of the nodes are different, this indicates a problem in the etcd cluster like partition.
+	// Do not perform below check as its result will differ by the member list used.
+	if !inconsistentMemberList{
+		for _, m := range memberList {
+			if _, ok := response[m.Name]; !ok {
+				// There are members in etcd cluster that is not part of controlplane nodes.
+				return response, errors.Errorf("etcd member %q is not part of the control plane nodes", m.Name)
+			}
```

Comment:
```suggestion
			// NOTE: we are using response for this check because it has a 1:1 correspondence with control plane nodes and map does support existence check
			if _, ok := response[m.Name]; !ok {
				return response, errors.Errorf("etcd member %q does not have a corresponding control plane nodes", m.Name)
			}
```